src/coreclr/vm/multicorejitimpl.h

```diff
@@ -20,84 +20,104 @@
+// Tags 0xfe0000 are currently free
 
-const int      MAX_RECORD_SIZE   = 2048;
-const unsigned MAX_JIT_COUNT     = (MAX_RECORD_SIZE - sizeof(unsigned)) / sizeof(unsigned);
+const unsigned RECORD_TYPE_OFFSET      = 24;        // offset of type of record
 
-const int      HEADER_W_COUNTER  = 14;              // Extra 16-bit counters in header for statistics: 28
-const int      HEADER_D_COUNTER  = 3;               // Extra 32-bit counters in header for statistics: 12
-const unsigned MAX_MODULES       = 512;             // Maximum number of modules
-
-const unsigned MAX_METHOD_ARRAY  = 16384;           // Maximum number of methods
-
-const int      MULTICOREJITLIFE  = 60 * 1000;       // 60 seconds
-
-const int      MULTICOREJITBLOCKLIMIT = 10 * 1000;  // 10 seconds
+const unsigned MAX_MODULES             = 0x10000;   // maximum allowed number of modules (2^16 values)
 
-const unsigned MAX_GENERIC_ARRAY  = 16384;          // Maximum number of generics
-                                                    //  8-bit module index
+const unsigned MODULE_LEVEL_OFFSET     = 16;        // offset of module load level
+const unsigned MAX_MODULE_LEVELS       = 0x100;     // maximum allowed number of module levels (2^8 values)
 
-                                                    // Method JIT information: 8-bit module 4-bit flag 20-bit method index
-const unsigned MODULE_DEPENDENCY = 0x800000;        //  1-bit module dependency mask
-const unsigned JIT_BY_APP_THREAD = 0x400000;        //  1-bit application thread
+const unsigned MAX_METHODS             = 0x4000;    // Maximum allowed number of methods (2^14 values) (in principle this is also limited by "unsigned short" counters)
 
-const unsigned METHODINDEX_MASK  = 0x0FFFFF;        // 20-bit method index
+const unsigned SIGNATURE_LENGTH_OFFSET = 16;        // offset of signature length
+const unsigned MAX_SIGNATURE_LENGTH    = 0x10000;   // maximum allowed signature length (2^16 values)
 
-                                                    // Dependendy information: 8-bit module 4-bit flag 4-bit unused 8-bit level 8-bit module
-const unsigned LEVEL_SHIFT       = 8;
-const unsigned LEVEL_MASK        = 0xFF;            //  8-bit file load level
-const unsigned MODULE_MASK       = 0xFF;            //  8-bit dependent module index
+const int      HEADER_W_COUNTER  = 14;              // Extra 16-bit counters in header for statistics: 28
+const int      HEADER_D_COUNTER  = 3;               // Extra 32-bit counters in header for statistics: 12
 
+const int      MULTICOREJITLIFE  = 60 * 1000;       // 60 seconds
 const int      MAX_WALKBACK      = 128;
 
-const unsigned SIGNATURELENGTH_MASK  = 0x0FFFFFF;        // 24-bit signature length
-
 enum
 {
-    MULTICOREJIT_PROFILE_VERSION   = 101,
+    MULTICOREJIT_PROFILE_VERSION   = 102,
 
+    // These should be powers of 2 in order to implement fast check for rec type
```

Comment:
Yes, what I wanted was to have simple bitfield check in RecorderInfo::IsMethod instead of shift and equality comparison. I agree that this is not critical at all, fixed